features/machine/machine.feature

```diff
@@ -69,3 +69,49 @@ Feature: Machine features testing
       | https://machine-api-operator.openshift-machine-api.svc:8443/metrics          | # @case_id OCP-25652
       | https://cluster-autoscaler-operator.openshift-machine-api.svc:9192/metrics   | # @case_id OCP-26111
       | https://machine-approver.openshift-cluster-machine-approver.svc:9192/metrics | # @case_id OCP-26102
+
+  # @author zhsun@redhat.com
+  # @case_id OCP-25608
+  @admin
+  @destructive
+  Scenario: Machine should have immutable field providerID and nodeRef
+    Given I have an IPI deployment
+    Given I store the last provisioned machine in the :machine clipboard
+
+    When I run the :get admin command with:
+      | resource      | machine                                |
+      | resource_name | <%= cb.machine %>                      |
+      | o             | yaml                                   |
+      | n             | openshift-machine-api                  |
+    Then the step should succeed
+    And evaluation of `@result[:parsed]["status"]["nodeRef"]["name"]` is stored in the :nodeRef_name clipboard
```

Comment:
@sunzhaohua2 I think line 82-87 can be replaced with ```machine(cb.machine).node_name```.  Looks like `providerID` may not be supported at the moment, but should be easy to add.

Reply:
@sunzhaohua2 You can update the file `lib/openshift/machine.rb`, define a method `provider_id` as the way `node_name` is defined.